src/sentry/api/endpoints/team_projects.py

```diff
@@ -118,37 +119,46 @@ def post(self, request, team):
         :param string name: the name for the new project.
         :param string slug: optionally a slug for the new project.  If it's
                             not provided a slug is generated from the name.
+        :param bool default_rules: create default rules (defaults to True)
         :auth: required
         """
         serializer = ProjectSerializer(data=request.data)
 
         if serializer.is_valid():
             result = serializer.validated_data
 
-            try:
-                with transaction.atomic():
-                    project = Project.objects.create(
-                        name=result["name"],
-                        slug=result.get("slug"),
-                        organization=team.organization,
-                        platform=result.get("platform"),
+            with transaction.atomic():
+                try:
+                    with transaction.atomic():
```

Comment:
I can remove it though the behavior will not quite be correct. It'd have to become:


```python
            try:
                with transaction.atomic():
                    project = Project.objects.create(
                        name=result["name"],
                        slug=result.get("slug"),
                        organization=team.organization,
                        platform=result.get("platform"),
                    )
                    project.add_team(team)

                    # XXX: create sample event?

                    self.create_audit_entry(
                        request=request,
                        organization=team.organization,
                        target_object=project.id,
                        event=AuditLogEntryEvent.PROJECT_ADD,
                        data=project.get_audit_log_data(),
                    )

                    project_created.send(
                        project=project,
                        user=request.user,
                        default_rules=result.get("default_rules", True),
                        sender=self,
                    )
            except IntegrityError:
                return Response(
                    {"detail": "A project with this slug already exists."}, status=409
                )
```

Which means if something other than Project create throws IntegrityError it will report the outcome incorrectly.